pkg/controller/jobs/pod/pod_controller_test.go

```diff
@@ -1359,7 +1358,7 @@ func TestReconciler(t *testing.T) {
 					KueueFinalizer().
 					Group("test-group").
 					GroupTotalCount("2").
-					StatusPhase(corev1.PodFailed).
+					StatusPhase(corev1.PodSucceeded).
```

Comment:
why change the status of these 2 pods? If not absolutely necessary, please revert.